src/Uno.Extensions.Hosting.UI/ApplicationBuilder.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.UI.Xaml;
+
+namespace Uno.Extensions.Hosting;
+
+internal class ApplicationBuilder : IApplicationBuilder
+{
+	private readonly List<Action<IHostBuilder>> _delegates = new List<Action<IHostBuilder>>();
+	public ApplicationBuilder(Application app, LaunchActivatedEventArgs arguments)
+	{
+		App = app;
+		Arguments = arguments;
+
+#if NET6_0_OR_GREATER && WINDOWS && !HAS_UNO
+		Window = new Window();
+#else
+        Window = Microsoft.UI.Xaml.Window.Current;
+#endif
+	}
+
+	public Application App { get; }
+	public LaunchActivatedEventArgs Arguments { get; }
+	public Window Window { get; }
+
+	public IHost Build()
+	{
+		var builder = UnoHost.CreateDefaultBuilder();
+		foreach (var @delegate in _delegates)
+		{
+			@delegate(builder);
+		}
+
+		// TODO: Need to expose the "enableUnoLogging" as part of UseLogging
+		return builder.Build();
```

Comment:
We now have https://github.com/unoplatform/uno.extensions/blob/a076d451484ce7e46a0577f730f7bcfa0ff2a683/src/Uno.Extensions.Hosting/IServiceInitialize.cs#L6 which didn't exist when we first wrote the startup logic.
We should extend UseLogging to specify the "enableUnoLogging" option and then use that to register an ISerivceInitialize instance which will do the same logic as calling Build(enabledUnoLogging: true)